.github/workflows/rust.yml

```
@@ -54,3 +56,29 @@ jobs:
     - run: cargo hack --feature-powerset check --target ${{ matrix.sys.target }}
     - if: matrix.sys.test
       run: cargo hack --feature-powerset test --target ${{ matrix.sys.target }}
+
+  cargo-publish:
+    if: github.event_name == 'release'
+    needs: [complete]
+    runs-on: ubuntu-latest
+    steps:
+    - if: contains(needs.*.result, 'failure')
+      run: exit 1
+    - uses: actions/checkout@v3
+    - run: rustup update
+    - run: cargo publish
```

Can you add the --frozen option? It should make sure the build process only uses what is in the lock file, so we know for sure the lock file is valid.
```suggestion
    - run: cargo publish --locked
```
The lock file will be discarded by crates.io, but it will jsut be a safeguard that the lock file we have at this moment is legit. We should probably add --locked to all the other jobs in here too.

---

you prefer `--frozen` or `--locked`?

---

(I _think_ `--frozen` won't work since that says "don't touch the network", so then .. `--locked`?)

---

I think frozen just won't touch the network to check, but it might still download, I'm not sure. Locked is probably sufficient.

---

Ok pushed locked everywhere.